### PR TITLE
Set defaultMaxMessageResults to 100 in DB adapters.

### PR DESCRIPTION
**Pull request by aforge**
Fetching more cannot will result in overwhelming the send queues
which effectively leads to session termination.

It's one of the problems that prevented us from running valid loadtests.

### CODE REVIEW

**Comment by aforge** [comment]
Right. Changed to use a separate maximum limit for messages.

File: server/db/mongodb/adapter.go [hunk 1]
```diff
@@ -138,6 +142,10 @@ func (a *adapter) Open(jsonconfig json.RawMessage) error {
 		a.maxResults = defaultMaxResults
 	}
 
+	if a.maxMessageResults <= 0 {
+		a.maxMessageResults = defaultMaxMessageResults
```
Comment (or-else) [inline server/db/mongodb/adapter.go:146]: `defaultMaxMessageResults` is not defined anywhere.

File: server/db/mongodb/adapter.go [hunk 2]
```diff
@@ -40,7 +43,8 @@ const (
 	adpVersion  = 111
 	adapterName = "mongodb"
 
-	defaultMaxResults = 1024
+	// This is capped by the Session's send queue limit (128).
+	defaultMaxResults = 100
```
Comment (or-else) [inline server/db/mongodb/adapter.go:47]: I think it should be left at 1024, while `defaultMaxMessageResults` should be defined as 100.

File: server/db/mongodb/adapter.go [hunk 3]
```diff
@@ -1955,7 +1963,7 @@ func (a *adapter) MessageDeleteList(topic string, toDel *t.DelMessage) error {
 
 // MessageGetDeleted returns a list of deleted message Ids.
 func (a *adapter) MessageGetDeleted(topic string, forUser t.Uid, opts *t.QueryOpt) ([]t.DelMessage, error) {
-	var limit = a.maxResults
+	var limit = a.maxMessageResults
```
Comment (or-else) [inline server/db/mongodb/adapter.go:1966]: It should be left as `maxResults`. Deleted ranges are collapsed into a single `meta` packet.

File: server/db/mysql/adapter.go [hunk 4]
```diff
@@ -2082,7 +2090,7 @@ var dellog struct {
 
 // Get ranges of deleted messages
 func (a *adapter) MessageGetDeleted(topic string, forUser t.Uid, opts *t.QueryOpt) ([]t.DelMessage, error) {
-	var limit = a.maxResults
+	var limit = a.maxMessageResults
```
Comment (or-else) [inline server/db/mysql/adapter.go:2093]: As before, please keep `a.maxResults`.

File: server/db/mongodb/adapter.go (hunk 2 above)
Comment (or-else) [reply server/db/mongodb/adapter.go:47]: BTW, why not keep it at a slightly higher value, like 112 or 120?

File: server/db/rethinkdb/adapter.go [hunk 5]
```diff
@@ -1711,7 +1720,7 @@ func (a *adapter) MessageGetAll(topic string, forUser t.Uid, opts *t.QueryOpt) (
 
 // Get ranges of deleted messages
 func (a *adapter) MessageGetDeleted(topic string, forUser t.Uid, opts *t.QueryOpt) ([]t.DelMessage, error) {
-	var limit = a.maxResults
+	var limit = a.maxMessageResults
```
Comment (or-else) [inline server/db/rethinkdb/adapter.go:1723]: As before, no need to change.

File: server/db/mongodb/adapter.go (hunk 2 above)
Comment (aforge) [reply server/db/mongodb/adapter.go]: Ah, sure. Thanks for catching. Apparently, my compile command does mysql adapter only.

File: server/db/mongodb/adapter.go (hunk 1 above)
Comment (aforge) [reply server/db/mongodb/adapter.go:147]: Done.

File: server/db/mongodb/adapter.go (hunk 3 above)
Comment (aforge) [reply server/db/mongodb/adapter.go]: Done.

File: server/db/rethinkdb/adapter.go (hunk 5 above)
Comment (aforge) [reply server/db/rethinkdb/adapter.go]: done.

File: server/db/mysql/adapter.go (hunk 4 above)
Comment (aforge) [reply server/db/mysql/adapter.go]: done.

**Pull request merged**